examples/full-example.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * This example demonstrates a simple commented workflow of:
+ * - initialising the RestClient and WebsocketClient for Gate.io exchange
+ * - connecting to an account’s private websockets (to receive updates asynchronously)
+ * - checking if connection is successful
+ * - fetching available futures balance
+ * - placing an order using 50% of the available balance
+ **/
+
+// Import the RestClient and WebsocketClient from the published version of this SDK, installed via NPM (npm install gateio-api)
+import { RestClient, WebsocketClient } from 'gateio-api';
+
+// Define the account object with API key and secret
+const account = {
+  // Replace 'yourApiHere' with your actual API key or use environment variables
+  key: process.env.API_KEY || 'yourApiHere',
+  // Replace 'yourSecretHere' with your actual API secret or use environment variables
+  secret: process.env.API_SECRET || 'yourSecretHere',
+};
+
+// Initialize the RestClient with the API credentials
+const gateRestClient = new RestClient({
+  apiKey: account.key,
+  apiSecret: account.secret,
+});
+
+// initialise websocket client - if you want only public data, you can initialise the client without the apiKey and apiSecret, just WebsocketClient()
+const gateWSClient = new WebsocketClient({
+  apiKey: account.key,
+  apiSecret: account.secret,
+});
+
+// Data received
+gateWSClient.on('update', (data) => {
+  console.info('data received: ', JSON.stringify(data));
+});
+
+async function subscribePrivateWs() {
+  try {
+    // Enter your user ID here
+    const myUserID = '20011';
+
+    //sub to balances updates
+    const userBalances = {
+      topic: 'futures.balances',
+      payload: [myUserID],
+    };
+
+    //sub to trades updates
+    const userTrades = {
+      topic: 'futures.usertrades',
+      payload: [myUserID, '!all'],
+    };
+
+    /**
+     * Either send one topic (with params) at a time
+     */
+    // client.subscribe({
+    //   topic: 'futures.usertrades',
+    //   payload: [myUserID, '!all'],
+    // }, 'perpFuturesUSDTV4');
+
+    /**
+     * Or send multiple topics in a batch (grouped by ws connection (WsKey))
+     * You can also use strings for topics that don't have any parameters, even if you mix multiple requests into one function call:
+     */
+    gateWSClient.subscribe([userBalances, userTrades], 'perpFuturesUSDTV4');
+
+    return true;
+  } catch (e) {
+    console.error(`Req error: `, e);
+    throw e;
+  }
+}
+
+async function main() {
+  try {
+    const isSubscribed = await subscribePrivateWs();
+    if (isSubscribed != true) {
+      console.log('Failed to subscribe to private ws');
+      return;
+    }
```

Comment:
```suggestion
await subscribePrivateWs();
```

Probably enough to do this, since it'll throw if there's an exception.